controlplane/pkg/nsm/nsm.go

```diff
@@ -295,6 +298,16 @@ func (srv *networkServiceManager) request(ctx context.Context, request networkse
 	// 11. We are done with configuration here.
 	logrus.Infof("NSM:(11-%v) Request done...", requestID)
 
+	if tools.GetConfig().SecurityProvider == nil {
+		logrus.Warnf("insecure: return connection without signature: %v", err)
+		return cc.GetConnectionSource(), nil
+	}
+
+	if err := security.SignConnection(
```

Comment:
The continuation of condition line 'merges' with body, 
please consider to extract the assignment out of 'if' statement.